scripts/update_function_docs/update_function_docs.go

```diff
@@ -0,0 +1,294 @@
+	fmt.Printf("%v\n", stdout)
+	return err
+}
+
+func gitShow() error {
+	stdout, err := runCmd("git", "show")
+	fmt.Printf("%v\n", stdout)
+	return err
+}
+
+var (
+	releaseBranchPattern = regexp.MustCompile(`[-\w]*\/(v\d*\.\d*)`)
+	releaseTagPattern    = regexp.MustCompile(`.*(go|ts)\/[-\w]*\/(v\d*\.\d*\.\d*)`)
+)
+
+func dirExists(path string) bool {
+	if stat, err := os.Stat(path); err == nil && stat.IsDir() {
+		return true
+	}
+	return false
+}
+
+type functionRelease struct {
+	FunctionName       string
+	MinorVersion       string
+	Language           string
+	LatestPatchVersion string
+	FunctionPath       string
+	ExamplePaths       []string
+}
+
+// newFunctionRelease allocates and initializes a functionRelease
+func newFunctionRelease(branch string) (*functionRelease, error) {
+	fr := &functionRelease{}
+	if !releaseBranchPattern.MatchString(branch) {
+		return nil, fmt.Errorf("invalid branch format")
+	}
+	segments := strings.Split(branch, "/")
+	fr.MinorVersion = segments[len(segments)-1]
+	fr.FunctionName = segments[len(segments)-2]
+	if err := fr.readLatestPatchVersion(); err != nil {
+		return nil, err
+	}
+	if err := fr.readDocPaths(); err != nil {
+		return nil, err
+	}
+	return fr, nil
+}
+
+// readLatestPatchVersion of the release from git tags
+func (fr *functionRelease) readLatestPatchVersion() error {
+	if fr.FunctionName == "" || fr.MinorVersion == "" {
+		return fmt.Errorf("missing function name and/or minor version")
+	}
+	tags, err := runCmd("git", "tag")
+	if err != nil {
+		return err
+	}
+	funcPattern := fmt.Sprintf("%s/%s", fr.FunctionName, fr.MinorVersion)
+	var lang, latestPatchVersion string
+	for _, tag := range strings.Split(tags, "\n") {
+		if !strings.Contains(tag, funcPattern) || !releaseTagPattern.MatchString(tag) {
+			continue
+		}
+		segments := strings.Split(tag, "/")
+		patchVersion := segments[len(segments)-1]
+		if latestPatchVersion == "" ||
+			semver.Compare(patchVersion, latestPatchVersion) == 1 {
+			latestPatchVersion = patchVersion
+			lang = segments[len(segments)-3]
+		}
+	}
+	if latestPatchVersion == "" || lang == "" {
+		return fmt.Errorf("could not find matching tag for release branch")
+	}
+	fr.Language = lang
+	fr.LatestPatchVersion = latestPatchVersion
+	return nil
+}
+
+// readDocPaths and set FunctionPath and ExamplePaths
+func (fr *functionRelease) readDocPaths() error {
+	executablePath, err := os.Executable()
+	if err != nil {
+		return err
+	}
+	repoBase := filepath.Dir(filepath.Dir(filepath.Dir(executablePath)))
+	fr.FunctionPath = filepath.Join(repoBase, "functions", fr.Language, fr.FunctionName)
+	examplesPath := filepath.Join(repoBase, "examples")
+
+	if !dirExists(fr.FunctionPath) {
+		fr.FunctionPath = filepath.Join(repoBase, "contrib", "functions", fr.Language, fr.FunctionName)
+		examplesPath = filepath.Join(repoBase, "contrib", "examples")
+		if !dirExists(fr.FunctionPath) {
+			return fmt.Errorf("function doc paths not found %s", fr.FunctionPath)
```

Comment:
This error message can be a little misleading, since it complains the contrib path only.

Reply:
Updated